server/src/main/java/org/opensearch/index/recovery/RemoteStoreRestoreService.java

```diff
@@ -233,6 +256,29 @@ private RemoteRestoreResult executeRestore(
         return RemoteRestoreResult.build(restoreUUID, restoreInfo, allocationService.reroute(updatedState, "restored from remote store"));
     }
 
+    private ClusterState restoreClusterMetadata(ClusterState restoredClusterState, Metadata clusterMetadata) {
```

Comment:
Do we need to handle data streams(index renaming etc) ?

Reply:
snapshot flow does that as we can rename the indices during restore. i think we should not need it here

Reply:
the variable names are confusing

Reply:
Also in executeRestore method, it is generating the cluster metadata. trying to understand what is the responsibility of each of them